### ISSUE TITLE
Panicking inside "replace_trailing_slash" function

### ISSUE THREAD

**Issue by shamblesides** [tokio-rs/axum#1122]
### Version
0.5.7

### Platform
Linux prod 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux

### Description
Axum is panicking here: https://github.com/tokio-rs/axum/blob/main/axum/src/routing/mod.rs#L532

```
panicked at 'called `Result::unwrap()` on an `Err` value: InvalidUriParts(InvalidUri(SchemeMissing))
```

**Comment by davidpdrsn** [comment]
I'm curious whether this is a bug in the `http` crate. I've asked in Discord.

**Comment by niklasf** [comment]
Was this bringing down the server, or was unwinding caught?

**Comment by davidpdrsn** [comment]
It wouldn't bring down the server. Just close the connection without sending a response.

The fix has been released in 0.5.10 🎉

**Comment by shamblesides** [comment]
FYI this was indeed bringing down my server, but I guess this is probably because I was aborting upon panicking
